integration/client/container_linux_test.go

```diff
@@ -1498,3 +1500,83 @@ func TestShimOOMScore(t *testing.T) {
 	case <-statusC:
 	}
 }
+
+// TestIssue9103 is used as regression case for issue 9103.
+//
+// The runc-fp will kill the init process so that the shim should return stopped
+// status after container.NewTask. It's used to simulate that the runc-init
+// might be killed by oom-kill.
+func TestIssue9103(t *testing.T) {
+	if os.Getenv("RUNC_FLAVOR") == "crun" {
+		t.Skip("skip it when using crun")
+	}
+	if getRuntimeVersion() == "v1" {
+		t.Skip("skip it when using shim v1")
+	}
```

Comment:
Added this (https://github.com/containerd/containerd/compare/2eee3df18421b95ccb7e5a750d1918a26d66b022..d0a1fedb5a4828daddff330a345780d0222e47e8)

thanks @fuweid (let's see if it goes green with this patch added); I'll update the 1.6 PR as well